### ISSUE TITLE
Jupyterlab Pull Request Template 

### ISSUE THREAD

**Issue by richagadgil** [jupyterlab/jupyterlab#5578]
Below I have a tentative Jupyterlab pull request template. The goal of this template is to make it easier to understand and assign pull requests to Jupyterlab. Every section has been written with the intent to have contributors provide as much information as possible and to encourage them to look at previous pull requests and opening issues before submitting new pull requests.

I invite any feedback and discussion on this!

@ellisonbg 

---

**What issue(s) does this pull request address?** Please look through [open pull requests](https://github.com/jupyterlab/jupyterlab/pulls) to see if someone has already started working on this. We encourage the community to open an issue describing proposed work before opening a pull request. 

**Describe your pull request**
Provide a clear and concise description of what your pull request does:

**Modified packages**
Provide a list of the JupyterLab packages your pull request modifies:

**Additional context**
Provide any other context about the pull request here:

**Version compatibility** [relevant sentence here]

**UI/UX Design Section:**

* **Expected behavior**
Provide a clear description of what visual and/or user interaction changes will occur as a result of your pull request:

* **Screenshots**
Attach screenshots of the original state and the new state. Include screenshots at different widths if applicable:

**Comment by blink1073** [comment]
Hi @richagadgil, thanks for kicking off the conversation!   I think all of our templates could use a commented out section for the intro so it doesn't render (conda-forge does this).  Perhaps even have a checklist the way they do.  cf https://github.com/conda-forge/jupyterlab-feedstock/tree/master/.github

**Comment by richagadgil** [comment]
I agree, it would be a lot cleaner with the intro commented out. A checklist would be useful to check for action items that should be completed prior to the pull request being submitted (running tests, checking version/build numbers). However, in this template we ask for information about the pull request rather than asking for certain actions to be completed. Is this something we want to include?

**Comment by blink1073** [comment]
Good point, we don't want to add to much to the template, let's keep it focused.
